### PR TITLE
Solve 3.2

### PR DESCRIPTION
**Pull request by IgorKruchinin**

### CODE REVIEW

**Comment by IgorKruchinin** [comment]
Последний Commit: https://github.com/IgorKruchinin/TKI-141/commit/d42821eeba18927814410027af4df461d434656a

**Comment by IgorKruchinin** [comment]
Поправил документацию: https://github.com/IgorKruchinin/TKI-141/commit/9483ecee74d34b1210d505f711b133a011efd0d2

File: task-4.3/main.c [hunk 1]
```diff
@@ -0,0 +1,219 @@
...
+#include "limits.h"
+#include "stdbool.h"
+#include "string.h"
+#include "math.h"
+
+/**
+* @brief печатает массив на экран
+* @param **arr исходный массив, n и m - количество строк и столбцов соответственно
+*/
+void printArr(int **arr, int n, int m);
```
Comment (MarinaPaley) [inline task-4.3/main.c:13]: `size_t`

File: task-4.3/main.c [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+int insertZeroColumns(int ***arr, int n, int m, bool *maxIndexes);
+
+/**
+* @brief выделяет память для массива
+* @param n и m размер массива
+* @return созданный массив
+*/
+int **initArray(size_t n, size_t m);
+
+int main() {
```
Comment (MarinaPaley) [inline task-4.3/main.c:68]: https://replit.com/@MarinaPaley/WebbedWobblyTabs

File: task-4.3/main.c [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+	int i = 0;
+	int j = 0;
+	int n = 0, m = 0, filling = 0;
+	puts("Enter shape of array (n and m): ");
+	if (scanf("%d %d", &n, &m) != 2) {
```
Comment (MarinaPaley) [inline task-4.3/main.c:73]: `int getInt("Введите ...");`

File: task-4.3/main.c [hunk 4]
```diff
@@ -0,0 +1,219 @@
...
+		puts("Invalid number of arguments!");
+		abort();
+	}
+	if (m <= 0 || n <= 0) {
```
Comment (MarinaPaley) [inline task-4.3/main.c:77]:
```c
size_t checkSize(int* size)
{
  if (size <= 0)
  {
    puts("Error");
    abort();
  }
returm (size_t)size;
}
```

}

File: task-4.3/main.c [hunk 5]
```diff
@@ -0,0 +1,219 @@
...
+	// Заменяем нулевой элемент иаксимальным по модулю элементом массива
+	puts("Output array:\n");
+	printArr(arr, n, m + insertedColumns);
+	return 0;
+}
+
+int *insert(int *arr, int size, int index, int elem) {
+	int *tmp = (int*)calloc(size + 1, sizeof(int));
+	memset(tmp, 0, size + 1);
+	int i = 0;
```
Comment (MarinaPaley) [inline task-4.3/main.c:110]: `size_t`

File: task-4.3/main.c [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+		}
+	}
+}
+
+void userInput(int ***arr, int n, int m) {
+	puts("Enter array elements:\n");
+	int i = 0, j = 0;
+	for (i = 0; i < n; ++i) {
+		for (j = 0; j < m; ++j) {
+				if (scanf("%d", &((*arr)[i][j])) != 1) {
```
Comment (MarinaPaley) [inline task-4.3/main.c:151]: `(*arr)[i][j]) = getInt()`

File: task-4.3/main.c [hunk 7]
```diff
@@ -0,0 +1,219 @@
...
+					puts("Invalid number of arguments!");
+					abort();
+				}
+		}
+		puts("\n");
+	}
+}
+
+int findMax(int ***arr, int n, int m) {
```
Comment (MarinaPaley) [inline task-4.3/main.c:160]: `const int** arr,...`

**Review by MarinaPaley — CHANGES_REQUESTED** [review]

**Pull request merged**